### PR TITLE
append cwd to gopath when debug program

### PR DESCRIPTION
**Pull request by muidea**
adjust current workspace path for debug path
set current workspace path to go path, like eclipse.
add cwd to debugger initialConfigurations

### CODE REVIEW

File: src/debugAdapter/goDebug.ts [hunk 1]
```diff
@@ -183,7 +187,7 @@ class Delve {
 				}
 			}
 			let dlvArgs = [mode || 'debug'];
-			if (mode === 'exec') {
+			if (mode === 'exec' || mode === 'debug') {
```
Comment (lukehoban) [inline src/debugAdapter/goDebug.ts]: Why is the program name being passed to `dlv debug`?  I don't believe it takes a program name as an argument.

File: src/debugAdapter/goDebug.ts [hunk 2]
```diff
@@ -196,15 +200,16 @@ class Delve {
...
+						
+			let dlvCwd = dirname(program);			
 			try {
 				if (lstatSync(program).isDirectory()) {
 					dlvCwd = program;
 				}
 			} catch (e) { }
+			
 			this.debugProcess = spawn(dlv, dlvArgs, {
-				cwd: dlvCwd,
```
Comment (lukehoban) [inline src/debugAdapter/goDebug.ts]: Is `dlvCwd` not being used anymore?

**Comment by lukehoban** [comment]
It's not clear to me that it's safe to assume that all users want their GOPATH to be modified in this way.

What is the specific issue that this is fixing?

**Comment by epsniff** [comment]
what's the problem this is trying to solve?   If it's running tests in subpackages, that can be solve with the `${file}` template directive in your debug config. 
```
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Testcase Debug",
			"type": "go",
			"request": "launch",
			"mode": "test",
			"program": "${file}",
			"env": {"TMPDIR": "${env.HOME}/tmp"},
			"args": [""]
		}
	]
}
```

**Comment by mattetti** [comment]
I agree with @epsniff, the issue OP is trying to address can be fixed in a way that wouldn't surprised programmers who don't expect their go path to be modified.

**Comment by ramya-rao-a** [comment]
@muidea Can you respond to the questions posed by @lukehoban @epsniff and @mattetti ?

**Comment by ramya-rao-a** [comment]
Closing this PR as we haven't heard back any answers from the OP to the questions posed.

**Pull request closed without merging**